src/boehm_gc.c

```diff
@@ -456,6 +456,7 @@ UInt ResizeBag(Bag bag, UInt new_size)
     CollectBags(0, 0);
 #endif
 
+    WriteGuard(bag);
```

Comment:
Doesn't `RetypeBag` need this as well?

Also, shouldn't it be inside an `#ifdef HPCGAP`?

Reply:
Sorry, this push went through prematurely. I've been experimenting with various checks on the header functions. `ResizeBag()` seems to not cause any problems; but other functions, such as `SIZE_BAG()`, do, when you add guards to them. This seems to be mostly related to some internal kernel objects, though, for reasons that I don't know yet.

If I add checks to `BAG_HEADER()` directly, this seems to cause an overhead on the order of 10%, possibly a bit less (with clang).